### DISCUSSION TITLE
vscode suggestion args problem

### DISCUSSION THREAD

**Discussion by hmz22** [rabbitmq/rabbitmq-java-client#1277] [Q&A] [answered]
I create maven app with vscode and when use this library show args badly like this:
`com.rabbitmq.client.Channel.exchangeDeclare(String arg0, String arg1, boolean arg2)`
show arg0 arg1 arg2 and I don't understand what is it
but must show like this:
`com.rabbitmq.client.Channel.exchangeDeclare(String exchangeName , String type , boolean durable )`
what is my problem in vscode?

**Comment by lukebakken** [comment, marked as answer]
Hello, thanks for using this library and RabbitMQ.

The problem you describe is not due to anything that this library can fix. It is due to how the Java language server you're using presents information to VSCode.

I use Neovim with the [`jdtls` language server](https://github.com/eclipse-jdtls/eclipse.jdt.ls). When I open a file in the [`rabbitmq-perf-test`](https://github.com/rabbitmq/rabbitmq-perf-test/blob/main/src/main/java/com/rabbitmq/perf/Utils.java) codebase, I see the same function signature in Neovim that you do in VSCode:

![image](https://github.com/rabbitmq/rabbitmq-java-client/assets/514926/8576b956-2712-469b-ac4e-9916b0aedead)

In your case, I recommend double-checking which Java language server you are using, and its configuration. You're probably using this plugin: https://github.com/redhat-developer/vscode-java

If you need further assistance, I suggest asking here: https://github.com/redhat-developer/vscode-java/discussions
